### PR TITLE
[Implement] isBuffer<T>(value: T)

### PR DESCRIPTION
**Pull request by jtenner**
This should be all set.

### CODE REVIEW

**Comment by jtenner** [comment]
@dcodeIO

```
> Buffer.isBuffer(null)
false
```

We need to do a null check, right?

**Comment by dcodeIO** [comment]
The `instanceof` operator should handle this if `T` is nullable.

**Comment by jtenner** [comment]
Alright. Then this should be all set

**Comment by jtenner** [comment]
Given that it's possible for `T` to be nullable, maybe this check is still okay? Unless if `T` is nullable it generates a null check too.

**Comment by dcodeIO** [comment]
If `T` is statically nullable, the `instanceof` will implicitly test for `null`.

**Comment by jtenner** [comment]
In that case we should be good. Will merge later.

File: tests/buffer.spec.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ describe("buffer", () => {
     expect<Buffer>(Buffer.alloc(10)).toHaveLength(10);
     let buff = Buffer.alloc(100);
     for (let i = 0; i < buff.length; i++) expect<u8>(buff[i]).toBe(0);
-    expect<ArrayBuffer>(buff.buffer).not.toBeNull();
+    expect<ArrayBuffer | null>(buff.buffer).not.toBeNull();
```
Comment (dcodeIO) [inline tests/buffer.spec.ts:30]: This looks like it can't happen

Comment (jtenner) [reply tests/buffer.spec.ts:30]: Remember that we call the `__alloc` function and set the property manually. This is just a sanity check to make sure that the `.buffer` property actually points to something.

Chances are it's something we won't forget, but I've forgotten way simpler things.

**Pull request merged**